File: payroll-api/src/main/java/com/company/payroll/mapper/StaffTrainingMapper.java
package com.company.payroll.mapper;

import java.util.List;

import com.company.payroll.model.StaffTraining;
import org.springframework.stereotype.Repository;

@Repository
public interface StaffTrainingMapper {
    int deleteByPrimaryKey(Integer tId);
    int insert(StaffTraining row);
    int insertSelective(StaffTraining row);
    List<StaffTraining> selectList();
    List<StaffTraining> selectListByStaffId(Integer staffId);
    StaffTraining selectByPrimaryKey(Integer tId);
    int updateByPrimaryKeySelective(StaffTraining row);
    int updateByPrimaryKeyWithBLOBs(StaffTraining row);
    int updateByPrimaryKey(StaffTraining row);

    /**
     * @deprecated will be removed after StaffTrainingMapper interface success implemented.
     * <p>Please use {@link #selectListByStaffId(Integer)} instead.</p>
     * @param mid
     * @return
     */
    @Deprecated
    List<StaffTraining> selectListByMId(int mid);
}

File: kotlin-payroll/src/main/kotlin/com/company/payroll/mapper/StaffTrainingMapper.kt
package com.company.payroll.mapper

import com.company.payroll.model.StaffTraining
import org.springframework.stereotype.Repository

@Repository
interface StaffTrainingMapper {
  fun deleteByPrimaryKey(tId: Int): Int

  fun insert(row: StaffTraining): Int

  fun insertSelective(row: StaffTraining): Int

  fun selectByPrimaryKey(tId: Int): StaffTraining

  fun selectList(): List<StaffTraining>

  fun selectListByStaffId(staffId: Int): List<StaffTraining>

  fun updateByPrimaryKeySelective(row: StaffTraining): Int

  fun updateByPrimaryKeyWithBLOBs(row: StaffTraining): Int

  fun updateByPrimaryKey(row: StaffTraining): Int
}